### PR TITLE
Allow tests to specify docker image

### PR DESCRIPTION
**Pull request by jgpruitt**
## Description

Some tests (upgrade tests) need to use specific docker images. Modifies the PostgresTestHarness to allow tests to specify the docker image to be used. Also changes the default docker image to the "local" image. When developing, this is most likely the image you want to be running tests against.

## Merge requirements

Please take into account the following non-code changes that you may need to make with your PR:

- [ ] CHANGELOG entry for user-facing changes
- [ ] Updated the relevant documentation

### CODE REVIEW

File: test-common/src/lib.rs [hunk 1]
```diff
@@ -23,29 +24,53 @@ impl PostgresTestHarness {
     const DB: &'static str = "postgres-db-test";
     const USER: &'static str = "postgres-user-test";
     const PASSWORD: &'static str = "postgres-password-test";
+    #[allow(dead_code)]
+    const IMAGE_URI_MASTER_BRANCH_14: &'static str =
+        "ghcr.io/timescale/dev_promscale_extension:master-ts2-pg14";
+    #[allow(dead_code)]
+    const IMAGE_URI_LOCAL_14: &'static str = "local/dev_promscale_extension:head-ts2-pg14";
```
Comment (sumerman) [inline test-common/src/lib.rs:31]: Do you need these constants for the upgrade tests or are they all for the sake of completeness?

Comment (jgpruitt) [reply test-common/src/lib.rs:31]: I think so. Either an env var will specify the image to use, OR we'll run with the default. The default is set to 14. If the env var is set to a v13 or v12, I need to detect it and use a "latest" image of the same version.

Comment (sumerman) [reply test-common/src/lib.rs:31]: I have a suggestion then. 

Firstly, you'd need to introduce two enums: 
```rust
enum ImageOrigin {
  Local,
  Master,
}

enum PgVersion {
  v14, v13, v12
}
```

Then you can have a function that returns an image URI based on these two parameters:
```rust
fn pg_image_uri(origin: ImageOrigin, version: PgVersion) -> String
```

Comment (jgpruitt) [reply test-common/src/lib.rs]: I like that. Updated

File: test-common/src/lib.rs [hunk 2]
```diff
@@ -27,25 +67,26 @@ impl PostgresTestHarness {
...
     }
 
     fn init_docker() -> Cli {
         clients::Cli::default()
     }
 
-    fn prepare_postgres_image() -> GenericImage {
-        images::generic::GenericImage::new(Self::postgres_image_uri()).with_wait_for(
+    fn prepare_image() -> GenericImage {
+        images::generic::GenericImage::new(Self::default_image_uri()).with_wait_for(
```
Comment (sumerman) [inline test-common/src/lib.rs:81]:
```suggestion
    fn prepare_image(&self) -> GenericImage {
        images::generic::GenericImage::new(self.img_uri).with_wait_for(
```

**Review by sumerman — CHANGES_REQUESTED** [review]
The part I'm requesting to change is around `prepare_image`. The rest is up to you.

File: test-common/src/lib.rs (hunk 2 above)
Comment (jgpruitt) [reply test-common/src/lib.rs:81]: oh yeah! whoops. good catch

File: test-common/src/lib.rs [hunk 3]
```diff
@@ -12,9 +12,49 @@ use testcontainers::{images, Container, Docker};
 /// A docker container running Postgres
 pub type PostgresContainer<'d> = Container<'d, Cli, GenericImage>;
 
+enum ImageOrigin {
```
[the remark this answers is not in the document]
Comment (jgpruitt) [reply test-common/src/lib.rs:15]: why is the allow dead_code necessary? We use the enums in the function which should mean it's not dead, right?

[the remark this answers is not in the document]
Comment (sumerman) [reply test-common/src/lib.rs:16]: IDK but when I was reviewing the PR I saw a bunch of linter complaints coming from CI

File: test-common/src/lib.rs [hunk 4]
```diff
@@ -27,25 +56,26 @@ impl PostgresTestHarness {
...
     }
 
     fn init_docker() -> Cli {
         clients::Cli::default()
     }
 
-    fn prepare_postgres_image() -> GenericImage {
-        images::generic::GenericImage::new(Self::postgres_image_uri()).with_wait_for(
+    fn prepare_image(&self) -> GenericImage {
+        images::generic::GenericImage::new(self.img_uri()).with_wait_for(
```
Comment (sumerman) [inline test-common/src/lib.rs:70]:
```suggestion
        images::generic::GenericImage::new(self.img_uri).with_wait_for(
```

File: test-common/src/lib.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (jgpruitt) [reply test-common/src/lib.rs:16]: ah yes. i see them too now.

File: test-common/src/lib.rs [hunk 5]
```diff
@@ -94,7 +133,7 @@ impl PostgresTestHarness {
     }
 
     pub fn run(&self) -> PostgresContainer {
-        let mut img = Self::prepare_postgres_image();
+        let mut img = Self::prepare_image();
```
[the remark this answers is not in the document]
Comment (jgpruitt) [reply test-common/src/lib.rs:136]: got it.

File: test-common/src/lib.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (jgpruitt) [reply test-common/src/lib.rs:17]: fixed

**Review by sumerman — APPROVED** [review]

**Pull request merged**